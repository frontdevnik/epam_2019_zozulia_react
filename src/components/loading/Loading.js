import React from 'react';
import styles from './loading.module.scss';

function Loading() {
  return (
    <div className={styles.loader}>
      <span>Loading...</span>
    </div>
  );
}

export default Loading;
