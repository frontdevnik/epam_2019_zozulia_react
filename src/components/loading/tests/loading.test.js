import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../Loading';
import styles from '../loading.module.scss';

describe('<Loading />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loading />);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should has main class', () => {
    expect(wrapper.find('div').hasClass(styles.loader)).toEqual(true);
  });

  it('should render text correctly', () => {
    expect(wrapper.find('span').text()).toEqual('Loading...');
  });
});
