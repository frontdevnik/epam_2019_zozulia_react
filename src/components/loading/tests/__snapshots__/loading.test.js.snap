// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render correctly 1`] = `"<div class=\\"loader\\"><span>Loading...</span></div>"`;
